openedx/core/djangoapps/credit/tests/test_api.py

```diff
@@ -480,16 +490,20 @@ def test_set_credit_requirement_status(self, mode):
         self.assert_grade_requirement_status(None, 0)
         credit_request.delete()
 
-        # Set the requirement to "satisfied" and check that it's actually set
-        api.set_credit_requirement_status(self.user, self.course_key, "grade", "grade")
-        self.assert_grade_requirement_status('satisfied', 0)
+        # order of below two assertions matter as:
+        # `failed` to `satisfied` is allowed
+        # `satisfied` to `failed` is not allowed
 
-        # Set the requirement to "failed" and check that it's actually set
+        # 1. Set the requirement to "failed" and check that it's actually set
         api.set_credit_requirement_status(self.user, self.course_key, "grade", "grade", status="failed")
         self.assert_grade_requirement_status('failed', 0)
 
+        # 2. Set the requirement to "satisfied" and check that it's actually set
+        api.set_credit_requirement_status(self.user, self.course_key, "grade", "grade")
+        self.assert_grade_requirement_status('satisfied', 0)
+
         req_status = api.get_credit_requirement_status(self.course_key, username)
-        self.assertEqual(req_status[0]["status"], "failed")
+        self.assertEqual(req_status[0]["status"], "satisfied")
```

Comment:
I don't think we can do it this way as at one time status would be either `failed` or `satisfied`.